redis/src/main/scala/zio/redis/ClusterExecutor.scala

```diff
@@ -58,8 +58,8 @@ final case class ClusterExecutor(
     }
 
     for {
-      key    <- ZIO.attempt(command(1)).orElseFail(CusterKeyError)
-      keySlot = Slot((key.asCRC16 % SlotsAmount).toLong)
+      keyOpt <- ZIO.succeed(command.respArgs.collectFirst { case key: RespArgument.Key => key })
+      keySlot = keyOpt.fold(Slot.Default)(key => Slot((key.asCRC16 % SlotsAmount).toLong))
```

Comment:
Since `SlotsAmount` is 2^14, you can replace `%` with `&`, i.e. `key.asCRC16 & (SlotsAmount - 1)`. The same applies for other usages of `%` with `SlotsAmount` (if any).